rust/crd/src/lib.rs

```diff
@@ -74,6 +78,14 @@ pub const DATA_DIR_NAME: &str = "/stackable/data";
 pub const KEYSTORE_DIR_NAME: &str = "/stackable/keystore";
 pub const USER_PASSWORD_DATA: &str = "/stackable/users";
 
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("object has no namespace associated"))]
+    NoNamespaceError,
+    #[snafu(display("Unknown Trino role found {}. Should be one of {:?}", role, roles))]
+    UnknownTrinoRole { role: String, roles: Vec<String> },
+}
```

Comment:
These are pretty unrelated errors, might be worth breaking them into separate types?

Reply:
I agree that they are unrelated but splitting this up (and we probably wont need the `UnknownTrinoRole` in the future) will result in another wrapper error in the controller.rs. Id like to avoid that for now.